src/structs/JPath.jl

```diff
@@ -0,0 +1,90 @@
+
+"""
+    JPath
+
+a polygon representation of a path, every Object will have a list of JPaths under the
+field jpaths.
+
+Every JPath has the following fields.
+
+- `polys` a list of polygons that makes up the path
+- `closed` a list of bools of same length as `polys`. closed[i] 
+states if polys[i] is a closed polygon or not
+- `fill` a vector of 4 numbers , R, G ,B and A the color it
+should be filled with. if the path was not filled its A is set to 0
+(along with R,G,B) this way its an "invisible" fill.
+- `stroke a vector of 4 numbers just like `fill` for the stroke color.
+- linewidth for stroke 
+"""
+mutable struct JPath
```

Comment:
I'm quite confused by the naming here. Every objects can have several `JPaths` but each `JPath` is also a vector of polys? For me a `JPath` sounds like a single path or polygon if closed.

Reply:
yes .
when a drawing function `obj.func` is called 
every call to strokepath / fillpath will add a new jpath.
so if an object.func does something like ..
`circle(O,100,:stroke)
line(O,100,:stroke)`
it will have 2 jpaths , one for the circle and one for the line.

lets say i instead had stored them as polygons . and i made some object like this .. a ring.
`circlepath(O,100,:path)
cirlcepath(O,50,:path,reversepath=true)
fillpath()
`
if i store every polygon and its associated stroke and fill , and i replay by placing every polygon and 
then immediatly stroke/fill i will not get a ring like the above code would draw.

Effectively an array JPath keeps track of where to call fill/stroke while drawing out the polygons inside it

Reply:
Thanks for the explanation. I think it makes more sense to me now. Would it make sense to call it `JPaths` though? As it holds information about several paths?

Reply:
maybe we should rename it alltogether.
i would prefer datastructures to have a "singular" name .
this allows little more clarity in name arrays of that structure.

for ex if a structure is called X
.. so arrays of it can be  Xs

but if its already called Xs , arrays of it can be a little confusing